File: DSA/DivyanshKumarSingh_590012175/Day_39_Question_1.c
#include <stdio.h>

#define MAX 100

int stack[MAX], top = -1;

void push(int x)
{
    stack[++top] = x;
}

int pop()
{
    return stack[top--];
}

int isEmpty()
{
    return top == -1;
}

void insertBottom(int x)
{
    if (isEmpty())
    {
        push(x);
        return;
    }
    int a = pop();
    insertBottom(x);
    push(a);
}

void reverse()
{
    if (isEmpty())
        return;
    int x = pop();
    reverse();
    insertBottom(x);
}

int main()
{
    int n, x;
    scanf("%d", &n);
    for (int i = 0; i < n; i++)
    {
        scanf("%d", &x);
        push(x);
    }
    reverse();
    for (int i = 0; i < n; i++)
        printf("%d ", stack[i]);
    return 0;
}

File: DSA/NiharikaChand_590017697/Day39Question1.java
import java.util.*;
class Day39Question1 
{
    public static void insert(Stack<Integer> stack, int el) 
    {
        if (stack.isEmpty()) 
        {
            stack.push(el);
            return;
        }

        int top = stack.pop();
        insert(stack, el);
        stack.push(top);
    }

    public static void reverse(Stack<Integer> stack) 
    {
        if (stack.isEmpty()) 
        {
            return;
        }
        int top = stack.pop();
        reverse(stack);
        insert(stack, top);
    }

    public static void main(String args[]) 
    {
        Scanner sc = new Scanner(System.in);
        Stack<Integer> stk = new Stack<>();

        System.out.println("Enter number of elements in the stack:");
        int n = sc.nextInt();

        System.out.println("Enter elements in top to bottom manner: ");
        for (int i = 0; i < n; i++) 
        {
            int num = sc.nextInt();
            stk.push(num);
        }

        reverse(stk);
        System.out.println("Output:");
        while (!stk.isEmpty()) 
        {
            System.out.print(stk.pop() + " ");
        }
    }
}

